solr/core/src/test/org/apache/solr/core/FSPRSTest.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.solr.common.NavigableObject;
+import org.apache.solr.common.cloud.DocCollection;
+import org.apache.solr.common.cloud.PerReplicaStates;
+import org.apache.solr.common.cloud.Replica;
+import org.apache.solr.common.cloud.ZkStateReader;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.common.util.Utils;
+
+public class FSPRSTest extends SolrCloudTestCase {
+
+
+  public void testIntegration() throws Exception {
+    String COLL = "prs_test_coll";
+    MiniSolrCloudCluster cluster =
+        configureCluster(3)
+            .withJettyConfig(jetty -> jetty.enableV2(true))
+            .addConfig("conf", configset("conf2"))
+            .configure();
+    try {
+      CollectionAdminRequest.createCollection(COLL, "conf", 1, 2)
+          .setPerReplicaState(Boolean.FALSE)
+          .process(cluster.getSolrClient());
+      cluster.waitForActiveCollection(COLL, 1, 2);
+
+      UpdateRequest ur = new UpdateRequest();
+      for(int i=0;i<10;i++) ur.add("id", ""+i);
+      ur.commit(cluster.getSolrClient(), COLL);
+
+      System.setProperty(CoreContainer.SOLR_QUERY_AGGREGATOR, "true");
+      JettySolrRunner qaJetty = cluster.startJettySolrRunner();
+      System.clearProperty(CoreContainer.SOLR_QUERY_AGGREGATOR);
+
+      try(HttpSolrClient client = (HttpSolrClient) qaJetty.newClient()){
+        NavigableObject result = (NavigableObject) Utils.executeGET(client.getHttpClient(),
+            qaJetty.getBaseUrl()+"/"+ COLL+"/select?q=*:*&wt=javabin", Utils.JAVABINCONSUMER
+            );
+        Collection<?> l = (Collection<?>) result._get("response", null);
+        assertEquals(10, l.size());
+      }
+      CollectionAdminRequest.modifyCollection(COLL,
+          Collections.singletonMap("perReplicaState", "true"))
+          .process(cluster.getSolrClient());
+      String collectionPath = ZkStateReader.getCollectionPath(COLL);
+      PerReplicaStates prs = PerReplicaStates.fetch(collectionPath, cluster.getZkClient(), null);
+      assertTrue(prs.states.size()>0);
+
+      CollectionAdminRequest.deleteReplica(COLL, "shard1",
+          cluster.getSolrClient().getClusterStateProvider().getCollection(COLL).getReplica((s, replica) -> !replica.isLeader()).getName()
+          ).process(cluster.getSolrClient());
+      cluster.waitForActiveCollection(COLL, 1, 1);
+      prs = PerReplicaStates.fetch(collectionPath, cluster.getZkClient(), null);
+      assertEquals(1, prs.states.size());
+
+
+      CollectionAdminRequest.SplitShard splitShard = CollectionAdminRequest.splitShard(COLL);
+      splitShard.setShardName("shard1");
+      NamedList<Object> response = splitShard.process(cluster.getSolrClient()).getResponse();
+      assertNotNull(response.get("success"));
+      for(int i= 0;i<100;i++) {
+        DocCollection docCollection = cluster.getSolrClient().getClusterStateProvider().getCollection(COLL);
+        int shardCount = docCollection.getSlices().size();
+        if(shardCount >=2) break;
```

Comment:
Looks like we need to delete original shard after split. Also please add assert for Queryaggregator active slices to two?